### ISSUE TITLE
[cegui] Fails to build 

### ISSUE THREAD

**Issue by jmdaemon** [acxz/pkgbuilds#160]
Using the AUR package (cegui 0.8.7-15) I get this build output
[cegui-build-output.txt](https://github.com/acxz/pkgbuilds/files/7765992/cegui-build-output.txt)

**Comment by acxz** [comment]
Sorry for the late response. The latest release for `cegui` is p old. Try to give `cegui-git` a shot. And let me know if that works for you.

**Comment by jmdaemon** [comment]
>Try to give `cegui-git` a shot

Does not seem to be working either.

```
-- Build files have been written to: /home/jmd/.cache/yay/cegui-git/src/cegui/build
[  0%] Building CXX object cegui/src/CMakeFiles/CEGUIBase-9999.dir/Affector.cpp.o
In file included from /home/jmd/.cache/yay/cegui-git/src/cegui/cegui/include/CEGUI/PropertySet.h:33,
                 from /home/jmd/.cache/yay/cegui-git/src/cegui/cegui/src/Affector.cpp:35:
/home/jmd/.cache/yay/cegui-git/src/cegui/cegui/include/CEGUI/IteratorBase.h:203:14: error: expected nested-name-specifier before ‘const’
  203 |     typename const ConstBaseIterator<T, typename T::mapped_type>::value_type&
      |              ^~~~~
/home/jmd/.cache/yay/cegui-git/src/cegui/cegui/include/CEGUI/IteratorBase.h:203:20: error: invalid use of qualified-name ‘CEGUI::ConstBaseIterator<T, typename T::mapped_type>::value_type’
  203 |     typename const ConstBaseIterator<T, typename T::mapped_type>::value_type&
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/jmd/.cache/yay/cegui-git/src/cegui/cegui/include/CEGUI/IteratorBase.h:203:67: error: expected ‘;’ at end of member declaration
  203 |     typename const ConstBaseIterator<T, typename T::mapped_type>::value_type&
      |                                                                   ^~~~~~~~~~
      |                                                                             ;
/home/jmd/.cache/yay/cegui-git/src/cegui/cegui/include/CEGUI/IteratorBase.h:204:5: error: ISO C++ forbids declaration of ‘getCurrentValue’ with no type [-fpermissive]
  204 |     getCurrentValue() const override
      |     ^~~~~~~~~~~~~~~
/home/jmd/.cache/yay/cegui-git/src/cegui/cegui/include/CEGUI/IteratorBase.h:294:14: error: expected nested-name-specifier before ‘const’
  294 |     typename const ConstBaseIterator<T>::value_type&
      |              ^~~~~
/home/jmd/.cache/yay/cegui-git/src/cegui/cegui/include/CEGUI/IteratorBase.h:294:20: error: invalid use of qualified-name ‘CEGUI::ConstBaseIterator<T>::value_type’
  294 |     typename const ConstBaseIterator<T>::value_type&
      |                    ^~~~~~~~~~~~~~~~~~~~
/home/jmd/.cache/yay/cegui-git/src/cegui/cegui/include/CEGUI/IteratorBase.h:294:42: error: expected ‘;’ at end of member declaration
  294 |     typename const ConstBaseIterator<T>::value_type&
      |                                          ^~~~~~~~~~
      |                                                    ;
/home/jmd/.cache/yay/cegui-git/src/cegui/cegui/include/CEGUI/IteratorBase.h:295:5: error: ISO C++ forbids declaration of ‘getCurrentValue’ with no type [-fpermissive]
  295 |     getCurrentValue() const override
      |     ^~~~~~~~~~~~~~~
make[2]: *** [cegui/src/CMakeFiles/CEGUIBase-9999.dir/build.make:76: cegui/src/CMakeFiles/CEGUIBase-9999.dir/Affector.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1038: cegui/src/CMakeFiles/CEGUIBase-9999.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
 -> error making: cegui-git

```

**Comment by acxz** [comment]
@jmdaemon sorry that I haven't been responding in a timely fashion. I don't use this package anymore and as such I don't actively build it all the time.

My current advice would be to post the error log you have gotten for the `-git` version of the package upstream: https://github.com/cegui/cegui/issues
and to refer the issue you create back to this issue chain so that we can keep track of it. I'm afraid the developers of cegui are your best hope. If I get some time (don't know when) I'll look into this.

**Comment by acxz** [comment]
Okay I found the reason of the error. Basically the `xerces-c` parser does not compile successfully. I have disabled it for now.

Patched with 776cc56
